runatlantis.io/docs/repo-level-atlantis-yaml.md

```diff
@@ -90,6 +90,44 @@ projects:
 This will stop Atlantis automatically running plan when `project1/` is updated
 in a pull request.
 
+Note:
+
+* When using `atlantis.yaml` to disable Autoplanning without listing other directories that should have Aitoplanning enabled, this will effectivly stop planning for all directories.
+
+Example:
+
+given the following project structure:
+```
+.
+├── project1
+│   └── main.tf
+└── project2
+    └── main.tf
+```
+if you add the following to `atlantis.yaml`:
+
+```yaml
+version: 3
+projects:
+- dir: project1
+  autoplan:
+    enabled: false
+```
+Autoplanning will not only stop for `project1`, rather Atlantis will ignore `project2` as well.
+
+So if you want to disable Autoplanning for `project1` and enable it for `project2` you have to explicitly state that and your config should be: 
+
+```yaml
+version: 3
+projects:
+- dir: project1
+  autoplan:
+    enabled: false
+- dir: project2
+  autoplan:
+    enabled: true
+```
+
```

Comment:
I think this sounds more like a bug.

base on this : 
https://www.runatlantis.io/docs/repo-level-atlantis-yaml.html#use-cases

it should only apply to the matched dir, I think we should file a bug instead

Reply:
I agree and it took me a while to understand how it works ... so until this is fixed I thought we could update the documentation to state clearly how to do achieve a directory level control on planning

Reply:
Agree. And until the bug is fixed I though it will help to have this behaviour documented clearly.

Reply:
I see the opposite behavior, where disabling it selectively on some directories keeps it working on dirs that don’t have it explicitly enabled or disabled.

Reply:
Interesting! 

Do you mind sharing your `atlantis.yaml`?

Reply:
Somewhat sanitized and simplified, but here's roughly how ours looks.

This handled the dependency issue (having plans run when a module a state / substate depends on gets changed, where I don't want to always autoplan, esp. on some really big states, or when Renovate is making PRs to update deps frequently), but does make it a little inconsistent, since now certain states autoplan and others don't.

Haven't tested yet if I can also just set `autoplan.enabled: false` at top level, which would make it more consistent.

```yaml
---
version: 3
#parallel_plan: true
projects:
  - dir: foo
    autoplan:
      enabled: false
      when_modified: ["*.tf*"]
  - dir: bar/baz
  - dir: meta/gcp/projects
[...]
  - dir: wizzle/thing1
    autoplan:
      enabled: false
      when_modified:
        - "*.tf*"
        - "../../modules/xyz/*.tf"
        - "../../modules/yyy/*.tf"
  - dir: wizzle/thing2
    autoplan:
      enabled: false
      when_modified:
        - "*.tf*"
        - "../../modules/othermodule/*.tf"
  - dir: wizzle/common
```

Reply:
Thanks for sharing. I will give it a try and will get back with the results I get.

Reply:
@wyardley version of atlantis?

Reply:
I am using `v0.19.2`